api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -500,20 +494,26 @@ impl KubernetesInfrastructure {
         &self,
         app_name: &str,
         service: &'a DeployableService,
-    ) -> Result<HashMap<&'a String, PersistentVolumeClaim>, KubernetesInfrastructureError> {
-        let client = self.client().await?;
+    ) -> Option<HashMap<&'a String, PersistentVolumeClaim>> {
+        let client = match self.client().await.ok() {
+            Some(val) => val,
+            None => {
+                return None;
+            }
+        };
 
-        let Runtime::Kubernetes(k8s_config) = self.config.runtime_config() else {return Ok(HashMap::new())};
+        let Runtime::Kubernetes(k8s_config) = self.config.runtime_config() else {return None};
```

Comment:
Isn't that the same and the better version?

```suggestion
        let Runtime::Kubernetes(k8s_config) = self.config.runtime_config()?;
```

Reply:
`?` is not valid here since `runtime_config()` returns `Runtime`
Also I don't think adding the else part is required here, `KubernetesRuntimeConfig` will either have config mapped value or a default value. I don't think there is a scenario where `KubernetesRuntimeConfig` or `Runtime` does not exist as they are always defaulted.
Please correct me if wrong.